compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1120,9 +1120,6 @@ public void visit(BLangXMLAttributeAccess xmlAttributeAccessExpr) {
         }
 
         checkExpr(indexExpr, env, symTable.stringType);
-        if (indexExpr.getKind() == NodeKind.XML_QNAME) {
```

Comment:
Are these unreached branches? Did we check why?

Reply:
Yeah, this is unreachable. The method returns from the if condition before this. Checked this with Supun.  Since `isUsedInXML` is `true` by default, there's no need for this here.